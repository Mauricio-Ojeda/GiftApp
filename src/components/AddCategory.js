import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')

    const handleInputChange = e => {
        setinputValue( e.target.value );

        console.log('llamado de handleInputChange');
    }

    const handleSubmit = e => {
        e.preventDefault();

        if( inputValue.trim().length > 1){
            setCategories(categories => [ inputValue, ...categories ]);
            setinputValue('');    
        }
    }

    return (
        <form onSubmit= { handleSubmit }>
            <input 
                type = "text"
                value = {inputValue}
                placeholder ='Busca tu Gift'
                onChange = { handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
  };

export default AddCategory


