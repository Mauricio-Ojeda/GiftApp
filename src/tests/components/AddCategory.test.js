import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';


describe('pruebas en AddCategory', () => {

    const setCategories = jest.fn();
    
    const wrapper = shallow( <AddCategory setCategories = { setCategories } />);

    test('debe de mostrar un snapshot del codigo html de Addcategory', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'hola Prisma'
        input.simulate('change', { target: { value } });

    })

    test('no debe postear la informacion si el input esta vacio', () => {
        
        wrapper.find('form').simulate('submit', { pereventDefault(){  }});

        expect( setCategories ).not.toHaveBeenColled();

    })
    
    
    

})
