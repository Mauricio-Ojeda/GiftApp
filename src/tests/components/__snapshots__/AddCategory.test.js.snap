// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en AddCategory debe de mostrar un snapshot del codigo html de Addcategory 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Busca tu Gift"
    type="text"
    value=""
  />
</form>
`;
